src/main/java/piano/main/PianoGUI.java

```diff
@@ -19,9 +23,14 @@ public PianoGUI(MidiChannel midiChannel, Recorder recorder) {
         root.setBackground(Color.BLACK);
 
         root.add(new RecorderPanel(recorder), BorderLayout.NORTH);
-        root.add(new Keyboard(midiChannel, recorder), BorderLayout.CENTER);
+        root.add(keyboard, BorderLayout.CENTER);
         root.add(new InstrumentsPanel(midiChannel), BorderLayout.SOUTH);
         setContentPane(root);
 
-	}
+        root.addKeyListener(new KeyboardListener(recorder, this));
+        root.setFocusable(true);
+    }
+    public Keyboard getKeyboard() {
+        return keyboard;
+    }
```

Comment:
It's this method necessary?

Reply:
In order to access the keyboard in KeyboardListener, yes. Unless I make the keyboard field public in PianoGUI.